main.go

```diff
@@ -335,31 +334,40 @@ func main() {
 						return err
 					}
 
-					var proof *prover.Proof
+					var r []byte
 					if mode == server.InsertionMode {
 						var params prover.InsertionParameters
 						err = json.Unmarshal(bytes, &params)
 						if err != nil {
 							return err
 						}
 						logging.Logger().Info().Msg("params read successfully")
-						proof, err = ps.ProveInsertion(&params)
+						var response *prover.InsertionResponse
+						response, err = ps.ProveInsertion(&params)
```

Comment:
this is weird – prove insertion needs the input hash and then you compute it again? something is duplicated